lib/iris/fileformats/grib/_load_convert.py

```diff
@@ -172,24 +174,24 @@ def _masker(item):
             result = result.data
     else:
         result = ma.masked
-        if value != _MDI and factor != _MDI:
+        if value not in _MDIs and factor not in _MDIs:
             result = _unscale(value, factor)
     return result
 
 
 # Regulations 92.1.4 and 92.1.5.
-_MDI = 2 ** 32 - 1
+_MDIs = [2 ** 31 - 1, 2 ** 32 - 1]
```

Comment:
yes, i confirm that this is explicitly needed for ECCodes integration.  the handling of missing values has changed.  I believe the new behaviour is more in line with the WMO regulations, where as the coding approach in current master is based on some interpretations made by gribapi which are not preserved in ECCodes